trustfall_core/src/interpreter/hints/tests/mod.rs

```diff
@@ -580,6 +581,31 @@ mod static_property_values {
         assert_eq!(adapter.on_starting_vertices.borrow()[&vid(1)].calls, 1);
     }
 
+    #[test]
+    fn required_properties_test() {
+        let input_name = "required_properties";
+
+        let adapter = TestAdapter {
+            on_starting_vertices: btreemap! {
+                vid(1) => TrackCalls::<ResolveInfoFn>::new_underlying(Box::new(|info| {
+                    assert!(info.coerced_to_type().is_none());
+                    assert_eq!(vid(1), info.vid());
+
+                    assert_eq!(hashset![
+                                    RequiredProperty::new("name".into()),
+                                    RequiredProperty::new("value".into()),
+                                    RequiredProperty::new("__typename".into())
+                                ], info.required_properties().collect::<HashSet<RequiredProperty>>());
```

Comment:
nit: can we collect into a `Vec` here to ensure we're also testing that the order of items remains unchanged?

I think for _external_ purposes, we shouldn't promise or document any particular order. But for our own test purposes, I think we want to stick to determinism so that snapshot tests remain easy and we don't get any flaky failures.